broadcaster/_backends/redis.py

```diff
@@ -12,10 +12,13 @@ def __init__(self, url: str):
         parsed_url = urlparse(url)
         self._host = parsed_url.hostname or "localhost"
         self._port = parsed_url.port or 6379
+        self.parse_query_db_password(parsed_url.query)
```

Comment:
Would it be cleaner if this method accepted the parsed_url and returned `host, port, db, password`. Then you can assign them all in `__init__` rather than assigning things inside a function inside `__init__`